topic_camera: "/dope/webcam/image_raw"
topic_camera_info: "/dope/webcam/camera_info"
topic_publishing: "dope"
input_is_rectified: True   # Whether the input image is rectified (strongly suggested!)
downscale_height: 400      # if the input image is larger than this, scale it down to this pixel height

# Comment any of these lines to prevent detection / pose estimation of that object
weights: {
    # "cracker":"package://dope/weights/cracker_60.pth",
    # "gelatin":"package://dope/weights/gelatin_60.pth",
    # "meat":"package://dope/weights/meat_20.pth",
    # "mustard":"package://dope/weights/mustard_60.pth",
    "soup":"package://dope/weights/soup_60.pth",
    #"sugar":"package://dope/weights/sugar_60.pth",
    # "bleach":"package://dope/weights/bleach_28_dr.pth"
    
    # NEW OBJECTS - HOPE
    # "AlphabetSoup":"package://dope/weights/AlphabetSoup.pth", 
    # "BBQSauce":"package://dope/weights/BBQSauce.pth", 
    # "Butter":"package://dope/weights/Butter.pth", 
    # "Cherries":"package://dope/weights/Cherries.pth", 
    # "ChocolatePudding":"package://dope/weights/ChocolatePudding.pth", 
    # "Cookies":"package://dope/weights/Cookies.pth", 
    # "Corn":"package://dope/weights/Corn.pth", 
    # "CreamCheese":"package://dope/weights/CreamCheese.pth", 
    # "GreenBeans":"package://dope/weights/GreenBeans.pth", 
    # "GranolaBars":"package://dope/weights/GranolaBars.pth", 
    # "Ketchup":"package://dope/weights/Ketchup.pth", 
    # "MacaroniAndCheese":"package://dope/weights/MacaroniAndCheese.pth", 
    # "Mayo":"package://dope/weights/Mayo.pth", 
    # "Milk":"package://dope/weights/Milk.pth", 
    # "Mushrooms":"package://dope/weights/Mushrooms.pth", 
    # "Mustard":"package://dope/weights/Mustard.pth", 
    # "Parmesan":"package://dope/weights/Parmesan.pth", 
    # "PeasAndCarrots":"package://dope/weights/PeasAndCarrots.pth",
    # "Peaches":"package://dope/weights/Peaches.pth",
    # "Pineapple":"package://dope/weights/Pineapple.pth",
    # "Popcorn":"package://dope/weights/Popcorn.pth",
    # "OrangeJuice":"package://dope/weights/OrangeJuice.pth", 
    # "Raisins":"package://dope/weights/Raisins.pth",
    # "SaladDressing":"package://dope/weights/SaladDressing.pth",
    # "Spaghetti":"package://dope/weights/Spaghetti.pth",
    # "TomatoSauce":"package://dope/weights/TomatoSauce.pth",
    # "Tuna":"package://dope/weights/Tuna.pth",
    # "Yogurt":"package://dope/weights/Yogurt.pth",

}

# Cuboid dimension in cm x,y,z
dimensions: {
    "cracker": [16.403600692749023,21.343700408935547,7.179999828338623],
    "gelatin": [8.918299674987793, 7.311500072479248, 2.9983000755310059],
    "meat": [10.164673805236816,8.3542995452880859,5.7600898742675781],
    "mustard": [9.6024150848388672,19.130100250244141,5.824894905090332],
    "soup": [6.7659378051757813,10.185500144958496,6.771425724029541],
    "sugar": [9.267730712890625,17.625339508056641,4.5134143829345703],
    "bleach": [10.267730712890625,26.625339508056641,7.5134143829345703],

    # new objects
    "AlphabetSoup" : [ 8.3555002212524414, 7.1121001243591309, 6.6055998802185059 ], 
    "Butter" : [ 5.282599925994873, 2.3935999870300293, 10.330100059509277 ], 
    "Ketchup" : [ 14.860799789428711, 4.3368000984191895, 6.4513998031616211 ],     
    "Pineapple" : [ 5.7623000144958496, 6.95989990234375, 6.567500114440918 ],
    "BBQSauce" : [ 14.832900047302246, 4.3478999137878418, 6.4632000923156738 ], 
    "MacaroniAndCheese" : [ 16.625600814819336, 4.0180997848510742, 12.350899696350098 ], 
    "Popcorn" : [ 8.4976997375488281, 3.825200080871582, 12.649200439453125 ],
    "Mayo" : [ 14.790200233459473, 4.1030998229980469, 6.4541001319885254 ], 
    "Raisins" : [ 12.317500114440918, 3.9751999378204346, 8.5874996185302734 ],
    "Cherries" : [ 5.8038997650146484, 7.0907998085021973, 6.6101999282836914 ], 
    "Milk" : [ 19.035800933837891, 7.326200008392334, 7.2154998779296875 ], 
    "SaladDressing" : [ 14.744099617004395, 4.3695998191833496, 6.403900146484375 ],
    "ChocolatePudding" : [ 4.947199821472168, 2.9923000335693359, 8.3498001098632812 ], 
    "Mushrooms" : [ 3.3322000503540039, 7.079899787902832, 6.5869998931884766 ], 
    "Spaghetti" : [ 4.9836997985839844, 2.8492999076843262, 24.988100051879883 ],
    "Cookies" : [ 16.724300384521484, 4.015200138092041, 12.274600028991699 ], 
    "Mustard" : [ 16.004999160766602, 4.8573999404907227, 6.5132999420166016 ], 
    "TomatoSauce" : [ 8.2847003936767578, 7.0198001861572266, 6.6469998359680176 ],
    "Corn" : [ 5.8038997650146484, 7.0907998085021973, 6.6101999282836914 ], 
    "OrangeJuice" : [ 19.248300552368164, 7.2781000137329102, 7.1582999229431152 ], 
    "Tuna" : [ 3.2571001052856445, 7.0805997848510742, 6.5837001800537109 ],
    "CreamCheese" : [ 5.3206000328063965, 2.4230999946594238, 10.359000205993652 ], 
    "Parmesan" : [ 10.286199569702148, 6.6093001365661621, 7.1117000579833984 ], 
    "Yogurt" : [ 5.3677000999450684, 6.7961997985839844, 6.7915000915527344 ],
    "GranolaBars" : [ 12.400600433349609, 3.8738000392913818, 16.53380012512207 ], 
    "Peaches" : [ 5.7781000137329102, 7.0961999893188477, 6.5925998687744141 ],
    "GreenBeans" : [ 5.758699893951416, 7.0608000755310059, 6.5732002258300781 ], 
    "PeasAndCarrots" : [ 5.8512001037597656, 7.0636000633239746, 6.5918002128601074 ] 
}

class_ids: {
    "cracker": 1,
    "gelatin": 2,
    "meat":    3,
    "mustard": 4,
    "soup":    5,
    "sugar":   6,
    "bleach":  7,
    "AlphabetSoup"      : 9, 
    "Ketchup"           : 10, 
    "Pineapple"         : 11,
    "BBQSauce"          : 12, 
    "MacaroniAndCheese" : 13, 
    "Popcorn"           : 14,
    "Butter"            : 15, 
    "Mayo"              : 16, 
    "Raisins"           : 17,
    "Cherries"          : 18, 
    "Milk"              : 19, 
    "SaladDressing"     : 20,
    "ChocolatePudding"  : 21, 
    "Mushrooms"         : 22, 
    "Spaghetti"         : 23,
    "Cookies"           : 24, 
    "Mustard"           : 25, 
    "TomatoSauce"       : 26,
    "Corn"              : 27, 
    "OrangeJuice"       : 28, 
    "Tuna"              : 29,
    "CreamCheese"       : 20, 
    "Parmesan"          : 31, 
    "Yogurt"            : 32,
    "GranolaBars"       : 33, 
    "Peaches"           : 34,
    "GreenBeans"        : 35, 
    "PeasAndCarrots"    : 36 
}

draw_colors: {
    "cracker": [13, 255, 128],  # green
    "gelatin": [255, 255, 255],  # while
    "meat": [0, 104, 255],  # blue
    "mustard": [217,12, 232],  # magenta
    "soup": [255, 101, 0],  # orange
    "sugar": [232, 222, 12],  # yellow
    "bleach": [232, 222, 12],  # yellow
}

# optional: provide a transform that is applied to the pose returned by DOPE
model_transforms: {
#    "cracker": [[ 0,  0,  1,  0],
#                [ 0, -1,  0,  0],
#                [ 1,  0,  0,  0],
#                [ 0,  0,  0,  1]]
}

# optional: if you provide a mesh of the object here, a mesh marker will be
# published for visualization in RViz
# You can use the nvdu_ycb tool to download the meshes: https://github.com/NVIDIA/Dataset_Utilities#nvdu_ycb
meshes: {
#    "cracker": "file://path/to/Dataset_Utilities/nvdu/data/ycb/aligned_cm/003_cracker_box/google_16k/textured.obj",
#    "gelatin": "file://path/to/Dataset_Utilities/nvdu/data/ycb/aligned_cm/009_gelatin_box/google_16k/textured.obj",
#    "meat":    "file://path/to/Dataset_Utilities/nvdu/data/ycb/aligned_cm/010_potted_meat_can/google_16k/textured.obj",
#    "mustard": "file://path/to/Dataset_Utilities/nvdu/data/ycb/aligned_cm/006_mustard_bottle/google_16k/textured.obj",
#    "soup":    "file://path/to/Dataset_Utilities/nvdu/data/ycb/aligned_cm/005_tomato_soup_can/google_16k/textured.obj",
#    "sugar":   "file://path/to/Dataset_Utilities/nvdu/data/ycb/aligned_cm/004_sugar_box/google_16k/textured.obj",
#    "bleach":  "file://path/to/Dataset_Utilities/nvdu/data/ycb/aligned_cm/021_bleach_cleanser/google_16k/textured.obj",
}

# optional: If the specified meshes are not in meters, provide a scale here (e.g. if the mesh is in centimeters, scale should be 0.01). default scale: 1.0.
mesh_scales: {
    "cracker": 0.01,
    "gelatin": 0.01,
    "meat":    0.01,
    "mustard": 0.01,
    "soup":    0.01,
    "sugar":   0.01,
    "bleach":  0.01,
}

overlay_belief_images: True   # Whether to overlay the input image on the belief images published on /dope/belief_[obj_name]

# Config params for DOPE
thresh_angle: 0.5
thresh_map: 0.01
sigma: 3
thresh_points: 0.1
